modules/auxiliary/gather/ip_geolocate.rb

```diff
@@ -0,0 +1,103 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'json'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Rex::Proto::Http
+  include Msf::Exploit::Remote::HttpClient
+  # HttpClient is only included because of normalize_uri
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'IP Based Geolocation',
+      'Description'    => %q{
+        This module uses a GeoIP API to locate the location
+        of a given IP address. Even though an external API is being
+        used, a API key is not needed for the module to work properly.
+        The shown longitude and latitude values are estimations.
+        If you want a more exact location, use the `wlan_geolocate` module.
+      },
+      'Author'         => [ 'Carter Brainerd <0xCB@protonmail.com>' ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptString.new("RHOSTS", [true, "A comma separated list of addresses to scan", nil])
```

Comment:
I wouldn't overload the meaning of `RHOSTS` in a module, especially when it's no longer `OptAddressRange`. Can you pick a less special name?